### PR TITLE
More EventGrid types

### PR DESCRIPTION
**Pull request by Porges**
Add a few different EventGrid types, also add Storage Queues to be used as subscription destinations.

### CODE REVIEW

File: v2/tools/generator/internal/codegen/pipeline/convert_allof_and_oneof_to_objects.go [hunk 1]
```diff
@@ -160,7 +160,63 @@ func (s synthesizer) getOneOfPropNames(oneOf *astmodel.OneOfType) ([]propertyNam
 		return err
 	})
 
-	return result, err
+	return simplifyPropNames(result), err
+}
+
+// simplifyPropNames makes the OneOf names better by trimming any common suffix
+func simplifyPropNames(names []propertyNames) []propertyNames {
```
[the remark this answers is not in the document]
Comment (Porges) [reply v2/tools/generator/internal/codegen/pipeline/convert_allof_and_oneof_to_objects.go]: Will do.

[the remark this answers is not in the document]
Comment (Porges) [reply v2/tools/generator/internal/codegen/pipeline/convert_allof_and_oneof_to_objects.go]: Done.

File: v2/config/samples/eventgrid/v1alpha1api20200601_eventsubscription.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: default
+spec:
+  owner:
+    name: sample-domain-topic
+  destination:
+    storageQueue:
+      endpointType: StorageQueue
+      properties: 
+        resourceReference: 
+          ARMID: "account-ARM-ID #TODO"
```
Comment (Porges) [inline v2/config/samples/eventgrid/v1alpha1api20200601_eventsubscription.yaml:14]: So @matthchr we don't have any other samples that use resource references yet…

Comment (Porges) [reply v2/config/samples/eventgrid/v1alpha1api20200601_eventsubscription.yaml]: Changed to use named resource reference

File: v2/azure-arm.yaml [hunk 3]
```diff
@@ -121,6 +121,16 @@ exportFilters:
...
     because: "including StorageAccountsBlobServicesContainer resource"
+  - action: include-transitive
+    group: storage
+    version: v*api20210401
+    name: 'StorageAccountsQueueService'
+    because: "including StorageAccountsQueueService resource"
+  - action: include-transitive
+    group: storage
+    version: v*api20210401
+    name: 'StorageAccountsQueueServices*'
```
Comment (babbageclunk) [inline v2/azure-arm.yaml:132]: If this is just adding the queue resource, why not specify it by name here? It made me suspicious. 🤨

File: v2/internal/controllers/crd_eventgrid_domain_test.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(rg),
+			Kind:     storage.StorageAccountsSpecKindStorageV2,
+			Sku:      storage.Sku{Name: storage.SkuNameStandardLRS},
+			// TODO: They mark this property as optional but actually it is required
+			AccessTier: &accessTier,
+		},
+	}
+
+	// TODO: Getting this is SUPER awkward
```
Comment (babbageclunk) [inline v2/internal/controllers/crd_eventgrid_domain_test.go:55]: You could try this with the k8s-level reference - it should be something like `acctReference := tc.MakeReferenceFromResource(acct)`. It should run with the same recording, right?

File: v2/internal/controllers/crd_eventgrid_domain_test.go [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		ObjectMeta: tc.MakeObjectMeta("dest-queue"),
+		Spec: storage.StorageAccountsQueueServicesQueues_Spec{
+			Owner: testcommon.AsOwner(queueServices),
+		},
+	}
+
+	tc.CreateResourceAndWait(queue)
+
+	armId := *domain.Status.Id
+	// objectKey := client.ObjectKeyFromObject(domain)
```
Comment (babbageclunk) [inline v2/internal/controllers/crd_eventgrid_domain_test.go:89]: May as well delete this.

File: v2/internal/controllers/crd_eventgrid_topic_test.go [hunk 6]
```diff
@@ -49,6 +52,80 @@ func Test_EventGrid_Topic(t *testing.T) {
...
+				queue := &storage.StorageAccountsQueueServicesQueue{
+					ObjectMeta: tc.MakeObjectMeta("queue"),
+					Spec: storage.StorageAccountsQueueServicesQueues_Spec{
+						Owner: testcommon.AsOwner(queueService),
+					},
+				}
+
+				tc.CreateResourceAndWait(queue)
+
+				// TODO: Getting this is SUPER awkward
```
Comment (babbageclunk) [inline v2/internal/controllers/crd_eventgrid_topic_test.go:95]: Same here.

**Review by babbageclunk — APPROVED** [review]
Looks good - try swapping out the armID references for k8s ones? Hmm - maybe it would cause a difference in the recording, since the referring resource wouldn't be able to send to arm until the referee had the ID back.

File: v2/azure-arm.yaml (hunk 3 above)
Comment (Porges) [reply v2/azure-arm.yaml:132]: it's both `StorageAccountsQueueServices` and `StorageAccountsQueueServicesQueue`

File: v2/internal/controllers/crd_eventgrid_domain_test.go (hunk 4 above)
Comment (Porges) [reply v2/internal/controllers/crd_eventgrid_domain_test.go:55]: Will try it and see…

Comment (Porges) [reply v2/internal/controllers/crd_eventgrid_domain_test.go]: it worked! 😁

**Pull request merged**